app/middleware/locals.js

```diff
@@ -1,3 +1,49 @@
+const constants = require('../lib/constants');
+
+function getLocationHeading(query) {
+  if (query.type) {
+    if ((query.type.localeCompare(constants.SERVICE_TYPES.professional, 'en', { sensitivity: 'base' }) === 0)
+      && ((query.origin === constants.SERVICE_CHOICES.symptoms)
+        || (query.origin === constants.SERVICE_CHOICES.under16)
+        || (query.origin === constants.SERVICE_CHOICES['16to25'])
+        || (query.origin === constants.SERVICE_CHOICES.over25))) {
+      return 'Where would you like to see a sexual health professional?';
+    }
+    if (query.type.localeCompare(constants.SERVICE_TYPES.kit, 'en', { sensitivity: 'base' }) === 0) {
+      if (query.origin === constants.SERVICE_CHOICES['16to25']) {
+        return 'Where would you like to collect your free test kit?';
+      } else if (query.origin === constants.SERVICE_CHOICES.over25) {
+        return 'Where would you like to collect your test kit?';
+      }
+    }
+  }
+  return undefined;
+}
+
+function mapServiceType(query) {
+  if (query.type) {
+    return query.type;
+  }
+  if (((query.age) && (query.age === constants.AGE.under16))
+    || ((query.symptoms) && (query.symptoms === constants.SYMPTOMS.yes))) {
+    return constants.SERVICE_TYPES.professional;
+  }
+  return undefined;
+}
+
+function mapServiceChoice(query) {
```

Comment:
Can this function be moved into a separate module?
It would facilitate some unit tests (I know there are integration tests already so it might be ok to leave as is) and means this module is just about setting locals.